### PR TITLE
Add file or directory tile to Dock persistent others

### PR DESCRIPTION
**Pull request by rmgpinto**
Added file or directory tile to Dock persistent others.
This will either expand directories or open files directly.

### CODE REVIEW

**Comment by Samasaur1** [comment]
It's

**Comment by rmgpinto** [comment]
@Samasaur1 I am trying to run the PR locally, but it doesn't generate the file `active-user`.
That's what I get in the logs.

**Comment by Samasaur1** [comment]
@rmgpinto What do you mean it "doesn't generate"? The logs give you the paths to two files in the Nix store — those are what I'm talking about

**Comment by Samasaur1** [comment]
You probably want to

```
opendiff /nix/store/4r7rk4742d0kpjivcqfksvnrh0apc6wb-system-defaults-write/activate-user.txt /nix/store/fx1rr7h100mavkwax300ckkyw0m72v9r-darwin-system-23.11pre558482.8a4282c38b6c+darwin4.7a694d1/activate-user
```
Find the block that is almost exactly the same. It should be pretty clear what the problem is

**Comment by rmgpinto** [comment]
@Samasaur1 thanks, it worked now.
It's ready to merge.

File: modules/system/defaults/dock.nix [hunk 1]
```diff
@@ -140,13 +140,13 @@ in {
...
       example = [ "~/Documents" "~/Downloads" ];
-      description = lib.mdDoc ''
+      description = ''
         Persistent folders in the dock.
       '';
       apply = value:
         if !(isList value)
         then value
-        else map (folder: { tile-data = { file-data = { _CFURLString = folder; _CFURLStringType = 15; }; }; }) value;
+        else map (folder: { tile-data = { file-data = { _CFURLString = "file://" + folder; _CFURLStringType = 15; }; }; tile-type = if strings.hasInfix "." folder then "file-tile" else "directory-tile"; }) value;
```
Comment (Enzime) [inline modules/system/defaults/dock.nix:149]: Folders can have `.` in their name like `~/.config`, ideally we should split by `/` then just look at the last part

**Review by Enzime — CHANGES_REQUESTED** [review]

File: modules/system/defaults/dock.nix (hunk 1 above)
Comment (rmgpinto) [reply modules/system/defaults/dock.nix]: You're right, I've pushed the changes.

**Comment by rmgpinto** [comment]
@Enzime ready with your proposed changes. Thanks

**Comment by Enzime** [comment]
Can you rebase over the latest master?

Everything else looks good to me :+1:

**Comment by rmgpinto** [comment]
done @Enzime

**Review by Enzime — APPROVED** [review]
LGTM

Thanks for the contribution

**Pull request merged**